### ISSUE TITLE
Inconsistent handling of method arguments when they are lambdas

### ISSUE THREAD

**Issue by JnRouvignac** [checkstyle/checkstyle#4638]
Indentation checks treat differently method arguments which are lambdas compared to function calls / literals / local variables / fields:

For example, if I modify the unit tests `InputIndentationLambda2` like this:
```java
public class InputIndentationLambda2 { //indent:0 exp:0
    public int literal() { //indent:4 exp:4
        return test(1, //indent:8 exp:8
       2); //indent:7 exp:7
    } //indent:4 exp:4

    public int localVariable() { //indent:4 exp:4
        int i = 0; //indent:8 exp:8
        return test(i, //indent:8 exp:8
       i); //indent:7 exp:7
    } //indent:4 exp:4

    private int i = 0; //indent:4 exp:4
    public int field() { //indent:4 exp:4
        return test(i, //indent:8 exp:8
       i); //indent:7 exp:7
    } //indent:4 exp:4

    public int method() { //indent:4 exp:4
        return test(f(), //indent:8 exp:8
       f()); //indent:7 exp:7
    } //indent:4 exp:4

    private int f() { //indent:4 exp:4
        return 0; //indent:8 exp:8
    } //indent:4 exp:4

    private int test(int f1, int f2) { //indent:4 exp:4
        return f2; //indent:8 exp:8
    } //indent:4 exp:4

    public Consumer<Integer> lambda() { //indent:4 exp:4
        return params((x) -> print(x * 1), //indent:8 exp:8
                      (x) -> print(x * 2)); //indent:22 exp:22
    } //indent:4 exp:4

   ... // rest of the existing code
```

The `IndentationCheck` fails with:

```java
java.lang.Error: Error was thrown while processing /home/jnrouvignac/git/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationLambda2.java
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:299)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:213)
	at com.puppycrawl.tools.checkstyle.BaseCheckTestSupport.verify(BaseCheckTestSupport.java:296)
	at com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheckTest.verify(IndentationCheckTest.java:163)
	at com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheckTest.verifyWarns(IndentationCheckTest.java:153)
	at com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheckTest.testLambda2(IndentationCheckTest.java:1700)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)
Caused by: java.lang.AssertionError: found a warning when none was expected for #0 at line 41 with message 'lambda arguments' has incorrect indentation level 22, expected level should be one of the following: 12, 16.
	at org.junit.Assert.fail(Assert.java:88)
	at com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheckTest$IndentAudit.addError(IndentationCheckTest.java:1885)
	at com.puppycrawl.tools.checkstyle.Checker.fireErrors(Checker.java:388)
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:287)
	... 28 more
```

I find it surprising:
1. It complains about the indentation for `lambda()`, but not for any of `literal()`, `localVariable()`, `field()` or `method()`. This is very inconsistent. It should complain for all or for none.
1. The suggested indentation for lambdas is wrong, 22 is totally fine in my opinion.

I can try to fix this issue, but I would like to know which indentation would be acceptable to you.
I would be fine if checkstyle ignored arguments which are lambdas, just like the other arguments.

**Comment by romani** [comment]
>The suggested indentation for lambdas is wrong, 22 is totally fine in my opinion.

It is line wrap, so it should be any indentation that is bigger than 8 + linewrapindentation

**Comment by JnRouvignac** [comment]
>> The suggested indentation for lambdas is wrong, 22 is totally fine in my opinion.

> It is line wrap, so it should be any indentation that is bigger than 8 + linewrapindentation

It is not bigger than xx characters.
Checkstyle insists that I move the code to the left, while I find it less readable.

> please report issue in format that we request
Understood. I will do that for sure.

Thank you!

**Comment by romani** [comment]
> Checkstyle insists that I move the code to the left, while I find it less readable.

http://checkstyle.sourceforge.net/config_misc.html#Indentation , make sure that `forceStrictCondition=false` in config to let check allow 
`indenation > base(8 in your case) + line-wrap-indentation`.

but you did not reported issue as required, so I can not give you exact advices

**Comment by plisker** [comment]
I've found the same error here. It seems that `forceStrictCondition` can be `false`, but acts as if `true` with lambda arguments

**Comment by plisker** [comment]
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/LambdaHandler.java#L77

There should be a `if (indentCheck.isForceStrictCondition())` check there, I believe

**Comment by romani** [comment]
@plisker , please send PR and lets have regression testing, here is tool that do most of work - https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#diffgroovy-diff-report-generation

**Comment by Abhishek-kumar09** [comment]
I have checked it on default indentation profile:
```
public class InputIndentationLambda2 { //indent:0 exp:0
    public int literal() { //indent:4 exp:4
        return test(1, //indent:8 exp:8
       2); //indent:7 exp:7
    } //indent:4 exp:4

    public int localVariable() { //indent:4 exp:4
        int i = 0; //indent:8 exp:8
        return test(i, //indent:8 exp:8
       i); //indent:7 exp:7
    } //indent:4 exp:4

    private int i = 0; //indent:4 exp:4
    public int field() { //indent:4 exp:4
        return test(i, //indent:8 exp:8
       i); //indent:7 exp:7
    } //indent:4 exp:4

    public int method() { //indent:4 exp:4
        return test(f(), //indent:8 exp:8
       f()); //indent:7 exp:7
    } //indent:4 exp:4

    private int f() { //indent:4 exp:4
        return 0; //indent:8 exp:8
    } //indent:4 exp:4

    private int test(int f1, int f2) { //indent:4 exp:4
        return f2; //indent:8 exp:8
    } //indent:4 exp:4

    public Consumer<Integer> lambda() { //indent:4 exp:4
        return params((x) -> print(x * 1), //indent:8 exp:8
                      (x) -> print(x * 2)); //indent:22 exp:22
    } //indent:4 exp:4
}
```
config:
```
<module name="Indentation" />
```
output:
```
java -jar checkstyle-8.36.2-all.jar -c config.xml Test.java 
Starting audit...
Audit done.
```
I think this should have been covered up in the scope of #8719
you had not provided the config.xml file, so I checked it with default configuration

**Comment by romani** [comment]
Let's close this issue.
Fixed in scope https://github.com/checkstyle/checkstyle/issues/3342
Released at 8.36.2
